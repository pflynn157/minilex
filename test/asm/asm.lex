
"mov" = Mov
"int" = Int
"syscall" = Syscall
"ret" = Ret

',' = Comma
':' = Colon
'.' = Dot

@comment_line = ;

