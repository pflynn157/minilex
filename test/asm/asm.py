base_path = "./base"
output_path = "./src"

keywords = [
    ("mov", "Mov"), ("int", "Int"), ("syscall", "Syscall"), ("ret", "Ret")
]

symbols = [
    (",", "Comma"), (":", "Colon"), (".", "Dot")
]

single_comments = [ ";" ]

multi_comments = []
