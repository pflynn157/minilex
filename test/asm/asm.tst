.data

.text

; Comment 2

_start:
    mov eax, 5
    mov ebx, 20
    int 0x20
    syscall
    ret ; Comment1

