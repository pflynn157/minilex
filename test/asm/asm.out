.
ID(data)
.
ID(text)
ID(_start)
:
mov
ID(eax)
,
INT(5)
mov
ID(ebx)
,
INT(20)
int
INT(32)
syscall
ret

